pilot/pkg/bootstrap/server.go

```diff
@@ -797,7 +801,21 @@ func (s *Server) initRegistryEventHandlers() {
 	s.ServiceController().AppendServiceHandler(serviceHandler)
 
 	if s.configController != nil {
-		configHandler := func(_, curr config.Config, event model.Event) {
+		configHandler := func(old config.Config, curr config.Config, event model.Event) {
+			if old.Generation == curr.Generation && curr.Generation != 0 {
+				// generation 0 indicates this is an ephemeral config generated by istio
```

Comment:
This comment wasn't very clear to me; it would have been more clear if we said Kubernetes will start generation at `1`, but some internally generated configurations may not set resource version at all and we still want updates from these.